src/internal/packager/images/push.go

```diff
@@ -134,13 +147,29 @@ func Push(ctx context.Context, cfg PushConfig) error {
 				return err
 			}
 
-			if err = pushImage(img, offlineName); err != nil {
+			err = retry.Do(
+				func() error { return pushImage(img, offlineName) },
+				retry.OnRetry(func(_ uint, err error) {
+					cfg.OCIConcurrency = 1
+					l.Debug("retrying image push", "error", err, "concurrency", cfg.OCIConcurrency)
+				}),
+				retry.Context(ctx),
+				retry.Attempts(2),
+				retry.Delay(500*time.Millisecond),
+			)
+			if err != nil {
 				return err
 			}
+
 			pushed = append(pushed, img)
 		}
 		return nil
-	}, retry.Context(ctx), retry.Attempts(uint(cfg.Retries)), retry.Delay(500*time.Millisecond))
+	}, retry.Context(ctx), retry.Attempts(uint(cfg.Retries)), retry.Delay(500*time.Millisecond), retry.OnRetry(func(attempt uint, _ error) {
+		if attempt == uint(cfg.Retries) {
+			cfg.ResponseHeaderTimeout = 60 * time.Second // this should really never happen
+		}
+		l.Debug("retrying component image(s) push", "response_timeout", cfg.ResponseHeaderTimeout)
+	}))
```

Comment:
This if statement will never be reached, it starts at attempt 0. Verified with below code
```go
func main() {
	retry.Do(func() error {
		return fmt.Errorf("error")
	}, retry.Attempts(3), retry.OnRetry(func(attempt uint, _ error) {
		fmt.Println("attempt", attempt)
		if attempt == uint(3) {
			fmt.Println("here")
		}
	}))
}
```

Reply:
verified fix with

```go
func main() {
	attempts := uint(3)

	retry.Do(func() error {
		fmt.Println("do")
		return fmt.Errorf("error")
	}, retry.Attempts(attempts), retry.OnRetry(func(attempt uint, _ error) {
		fmt.Println("on retry", attempt)
		if attempts > 2 && attempt == attempts-2 {
			fmt.Println("last try")
		}
	}))
}
```